pkg/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,176 @@
+package bootstrap
+
+import (
+	"context"
+
+	"github.com/solo-io/skv2/pkg/stats"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/spf13/pflag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/go-logr/zapr"
+	"github.com/solo-io/go-utils/contextutils"
+	v1 "github.com/solo-io/skv2/pkg/api/core.skv2.solo.io/v1"
+	"github.com/solo-io/skv2/pkg/multicluster"
+	"github.com/solo-io/skv2/pkg/multicluster/watch"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	zaputil "sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	// required import to enable kube client-go auth plugins
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+)
+
+// StartParameters specify paramters for starting a generic controller which may need access to its local cluster as well as remote (multicluster) clients and managers
+type StartParameters struct {
+	Ctx             context.Context
```

Comment:
nit: can we keep the `context` out of the struct. I personally find it much more complicated to track context when it is not directly passed from function to function.